File: utillib/src/main/java/com/saint/util/util/toast/AppToast.java
package com.saint.util.util.toast;


import android.view.Gravity;

import com.blankj.utilcode.util.ConvertUtils;
import com.blankj.utilcode.util.ToastUtils;
import com.saint.util.UtilConfig;
import com.saint.util.util.AppUtil;

/**
 *
 */

public final class AppToast {

    public static void tShort(CharSequence content) {
        ToastUtils.showShort(content);
    }

    public static void tShort(int contentRes) {
        ToastUtils.showShort(contentRes);
    }

    public static void tLong(CharSequence content) {
        ToastUtils.showLong(content);
    }

    public static void tLong(int contentRes) {
        ToastUtils.showLong(contentRes);
    }

    public static void showAtTop(CharSequence msg) {
        ToastUtils.setGravity(Gravity.TOP, 0, 0);
        ToastUtils.showShort(msg);
    }

    public static void showAtTop(int msg) {
        ToastUtils.setGravity(Gravity.TOP, 0, 0);
        ToastUtils.showShort(msg);
    }

    public static void showInCenter(CharSequence msg) {
        ToastUtils.setGravity(Gravity.CENTER, 0, 0);
        ToastUtils.showShort(msg);
    }

    public static void showInCenter(int msg) {
        ToastUtils.setGravity(Gravity.CENTER, 0, 0);
        ToastUtils.showShort(msg);
    }

    public static void showAtLocation(CharSequence msg, int gravity, float xOffsetDp, float yOffsetDp) {
        ToastUtils.setGravity(gravity, ConvertUtils.dp2px(xOffsetDp), ConvertUtils.dp2px(yOffsetDp));
        ToastUtils.showShort(msg);
    }

    public static void showAtLocation(int msg, int gravity, float xOffsetDp, float yOffsetDp) {
        ToastUtils.setGravity(gravity, ConvertUtils.dp2px(xOffsetDp), ConvertUtils.dp2px(yOffsetDp));
        ToastUtils.showShort(msg);
    }

    public static void resetToast() {
        ToastUtils.setMsgColor(-0x1000001);
        ToastUtils.setBgColor(-0x1000001);
        ToastUtils.setBgResource(-1);
        ToastUtils.setGravity(Gravity.CENTER, -1, -1);
    }
}

File: utillib/src/main/java/com/saint/util/util/toast/AppToast.kt
package com.saint.util.util.toast

import android.view.Gravity
import com.hjq.toast.Toaster
import com.saint.util.UtilConfig
import com.saint.util.util.AppUtil

/**
 *
 */
object AppToast {

    init {
        if (!Toaster.isInit()) {
            Toaster.init(UtilConfig.getApp())
        }
    }

    fun show(content: CharSequence?) {
        Toaster.show(content)
    }

    fun show(contentRes: Int) {
        Toaster.show(contentRes)
    }

    fun tShort(content: CharSequence?) {
        Toaster.showShort(content)
    }

    fun tShort(contentRes: Int) {
        Toaster.showShort(contentRes)
    }

    fun tLong(content: CharSequence?) {
        Toaster.showLong(content)
    }

    fun tLong(contentRes: Int) {
        Toaster.showLong(contentRes)
    }

    fun showAtTop(msg: CharSequence?) {
        Toaster.setGravity(Gravity.TOP, 0, 0)
        Toaster.show(msg)
    }

    fun showAtTop(msg: Int) {
        Toaster.setGravity(Gravity.TOP, 0, 0)
        Toaster.show(msg)
    }

    fun showInCenter(msg: CharSequence?) {
        Toaster.setGravity(Gravity.CENTER, 0, 0)
        Toaster.show(msg)
    }

    fun showInCenter(msg: Int) {
        Toaster.setGravity(Gravity.CENTER, 0, 0)
        Toaster.show(msg)
    }

    fun showAtLocation(msg: CharSequence?, gravity: Int, xOffsetDp: Float, yOffsetDp: Float) {
        Toaster.setGravity(gravity, AppUtil.dpToPx(xOffsetDp), AppUtil.dpToPx(yOffsetDp))
        Toaster.show(msg)
    }

    fun showAtLocation(msg: Int, gravity: Int, xOffsetDp: Float, yOffsetDp: Float) {
        Toaster.setGravity(gravity, AppUtil.dpToPx(xOffsetDp), AppUtil.dpToPx(yOffsetDp))
        Toaster.show(msg)
    }
}